src/EditorFeatures/CSharpTest/EditAndContinue/TopLevelEditingTests.cs

```diff
@@ -6737,6 +7081,180 @@ public void IndexerWithExpressionBody_Update()
+            });
+        }
+
+        [Fact, WorkItem(17681, "https://github.com/dotnet/roslyn/issues/17681")]
+        public void Indexer_BlockBodyToGetterExpressionBody()
+        {
+            var src1 = "class C { int this[int a] { get { return 1; } } }";
+            var src2 = "class C { int this[int a] { get => 1; } }";
+            var edits = GetTopEdits(src1, src2);
+
+            edits.VerifyEdits("Update [get { return 1; }]@28 -> [get => 1;]@28");
+            edits.VerifyRudeDiagnostics();
+        }
+
+        [Fact, WorkItem(17681, "https://github.com/dotnet/roslyn/issues/17681")]
+        public void Indexer_GetterExpressionBodyToExpressionBody()
+        {
+            var src1 = "class C { int this[int a] { get => 1; } }";
+            var src2 = "class C { int this[int a] => 1; } ";
+
+            var edits = GetTopEdits(src1, src2);
+
+            edits.VerifyEdits(
+                "Update [int this[int a] { get => 1; }]@10 -> [int this[int a] => 1;]@10",
+                "Delete [{ get => 1; }]@26",
+                "Delete [get => 1;]@28");
+
+            edits.VerifySemantics(ActiveStatementsDescription.Empty, new[]
+            {
+                SemanticEdit(SemanticEditKind.Update, c => c.GetMember("C.get_Item"), preserveLocalVariables: false)
+            });
+        }
+
+        [Fact, WorkItem(17681, "https://github.com/dotnet/roslyn/issues/17681")]
+        public void Indexer_ExpressionBodyToGetterExpressionBody()
+        {
+            var src1 = "class C { int this[int a] => 1; }";
+            var src2 = "class C { int this[int a] { get => 1; } }";
+            var edits = GetTopEdits(src1, src2);
+
+            edits.VerifyEdits(
+                "Update [int this[int a] => 1;]@10 -> [int this[int a] { get => 1; }]@10",
+                "Insert [{ get => 1; }]@26",
+                "Insert [get => 1;]@28");
+
+            edits.VerifySemantics(ActiveStatementsDescription.Empty, new[]
+            {
+                SemanticEdit(SemanticEditKind.Update, c => c.GetMember("C.get_Item"), preserveLocalVariables: false)
+            });
+        }
+
+        [Fact, WorkItem(17681, "https://github.com/dotnet/roslyn/issues/17681")]
+        public void Indexer_GetterBlockBodyToGetterExpressionBody()
+        {
+            var src1 = "class C { int this[int a] { get { return 1; } set { Console.WriteLine(0); } } }";
+            var src2 = "class C { int this[int a] { get => 1; set { Console.WriteLine(0); } } }";
+
+            var edits = GetTopEdits(src1, src2);
+
+            edits.VerifyEdits("Update [get { return 1; }]@28 -> [get => 1;]@28");
+
+            edits.VerifySemantics(ActiveStatementsDescription.Empty, new[]
+            {
+                SemanticEdit(SemanticEditKind.Update, c => c.GetMember("C.get_Item"), preserveLocalVariables: false),
+                SemanticEdit(SemanticEditKind.Update, c => c.GetMember("C.set_Item"), preserveLocalVariables: false)
```

Comment:
ok. let us try

---
In reply to: [128833161](https://github.com/dotnet/roslyn/pull/20931#discussion_r128833161) [](ancestors = 128833161)